compilation of packages on Windows can fail if RTOOLS40_HOME unset

### System details

    RStudio Edition : Desktop
    RStudio Version : any
    OS Version      : Windows 10 x64
    R Version       : 4.0.0

### Steps to reproduce the problem

Use RStudio to install Rtools; e.g. via

```
if (!file.exists("C:/RBuildTools/4.0"))
  .rs.installBuildTools("Installing Rtools")
```

Then, put that on the PATH:

```
path <- Sys.getenv("PATH")
Sys.setenv(PATH = paste("C:\\RBuildTools\\4.0\\usr\\bin", path, sep = ";"))
```

Then, ensure that `RTOOLS40_HOME` is unset; e.g.

```
Sys.unsetenv("RTOOLS40_HOME")
```

Finally, try to install a package from sources that requires compilation, e.g.

```
install.packages("digest", type = "source")
```

### Describe the problem in detail

Installation fails as the wrong path to Rtools40 is used:

```
> install.packages("digest", type = "source")
Installing package into 'C:/Users/kevinushey/r/win-library/4.0'
(as 'lib' is unspecified)
trying URL 'https://cran.rstudio.com/src/contrib/digest_0.6.25.tar.gz'
Content length 145642 bytes (142 KB)
downloaded 142 KB

* installing *source* package 'digest' ...
** package 'digest' successfully unpacked and MD5 sums checked
** using staged installation
** libs

*** arch - i386
"c:/rtools40/mingw32/bin/"g++  -std=gnu++11 -I"C:/R/R-40~1.0/include" -DNDEBUG     -I"C:/Users/kevinushey/AppData/Local/renv/extsoft/include"     -O2 -Wall  -mfpmath=sse -msse2 -mstackrealign -c SpookyV2.cpp -o SpookyV2.o
sh: c:/rtools40/mingw32/bin/g++: No such file or directory
make: *** [C:/R/R-40~1.0/etc/i386/Makeconf:229: SpookyV2.o] Error 127
ERROR: compilation failed for package 'digest'
* removing 'C:/Users/kevinushey/r/win-library/4.0/digest'
* restoring previous 'C:/Users/kevinushey/r/win-library/4.0/digest'
Warning in install.packages :
  installation of package 'digest' had non-zero exit status

The downloaded source packages are in
	'C:\tmp\RtmpemEXb7\downloaded_packages'
```

### Describe the behavior you expected

The correct path to the compiler tools should be used.

- [x] I have read the guide to submitting good bug reports at https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports . 
- [x] I have installed the latest version of RStudio and confirmed that the issue still persists.
- [x] If I am reporting a RStudio crash, I have included a diagnostics report. https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

--- a/src/cpp/core/r_util/RToolsInfo.cpp
+++ b/src/cpp/core/r_util/RToolsInfo.cpp
@@ -198,6 +198,11 @@ RToolsInfo::RToolsInfo(const std::string& name,
       // set BINPREF
       environmentVars.push_back({"BINPREF", "/mingw$(WIN)/bin/"});
 
+      // set RTOOLS40_HOME
+      std::string rtoolsPath = installPath.getAbsolutePath();
+      std::replace(rtoolsPath.begin(), rtoolsPath.end(), '\\', '/');
+      environmentVars.push_back({"RTOOLS40_HOME", rtoolsPath});
+
       // set clang args
 #ifdef _WIN64
       std::string baseDir = "mingw64";

